### PR TITLE
RD-1400 - add domains scrubbing

### PR DESCRIPTION
**Pull request by saartochner**

### CODE REVIEW

File: src/lumigo_tracer/utils.py [hunk 1]
```diff
@@ -55,6 +62,7 @@ def config(
     :param enhance_print: Should we add prefix to the print (so the logs will be in the platform).
     :param step_function: Is this function is a part of a step function?
     :param timeout_timer: Should we start a timer to send the traced data before timeout acceded.
+    :param domains_scrubber: List of regexex. We will not collect data of requests with hosts that match it.
```
Comment (nirLumigo) [inline src/lumigo_tracer/utils.py]:
```suggestion
    :param domains_scrubber: List of regexes. We will not collect data of requests with hosts that match it.
```

File: src/lumigo_tracer/utils.py [hunk 2]
```diff
@@ -67,6 +75,16 @@ def config(
...
     Configuration.send_only_if_error = os.environ.get("SEND_ONLY_IF_ERROR", "").lower() == "true"
+    if "LUMIGO_DOMAINS_SCRUBBER" in os.environ:
+        try:
+            Configuration.domains_scrubber = json.loads(os.environ["LUMIGO_DOMAIN_SCRUBBER"])
+        except Exception:
+            get_logger().critical(
+                "Could not parse the specified domains scrubber, shutting down the reporter."
+            )
+            Configuration.should_report = False
+    else:
```
Comment (nirLumigo) [inline src/lumigo_tracer/utils.py]: Shouldn't it be the opposite in the if-else? Don't we prefer that the argument will override the environment variable?

File: src/test/unit/sync_http/test_sync_hook.py [hunk 3]
```diff
@@ -195,6 +195,30 @@ def lambda_test_function():
...
+    def lambda_test_function():
+        return http.client.HTTPConnection(host="www.google.com").send(b"\r\n")
+
+    lambda_test_function()
+    http_events = SpansContainer.get_span().http_spans
+    assert len(http_events) == 1
+    assert "google" in http_events[0].get("info", {}).get("httpInfo", {}).get("host")
+    assert "headers" not in http_events[0]["info"]["httpInfo"]["request"]
+    assert http_events[0]["info"]["httpInfo"]["request"]["body"] == "The data is not available"
+
```
Comment (nirLumigo) [inline src/test/unit/sync_http/test_sync_hook.py]: Add a test with `domain_scrubber` but send a request to a website that is not supposed to be scrubbed, and make sure the span collected as before (not scrubbed)

File: src/lumigo_tracer/utils.py [hunk 4]
```diff
@@ -67,6 +75,16 @@ def config(
     Configuration.is_step_function = step_function
     Configuration.timeout_timer = timeout_timer
     Configuration.send_only_if_error = os.environ.get("SEND_ONLY_IF_ERROR", "").lower() == "true"
+    if "LUMIGO_DOMAINS_SCRUBBER" in os.environ:
+        try:
+            Configuration.domains_scrubber = json.loads(os.environ["LUMIGO_DOMAIN_SCRUBBER"])
```
Comment (nirLumigo) [inline src/lumigo_tracer/utils.py]: Are you sure this is in json format?

File: README.md [hunk 5]
```diff
@@ -20,6 +20,7 @@ def my_lambda(event, context):
 ## Configuration
 * You can turn on the debug logs by setting the environment variable `LUMIGO_DEBUG=true`
 * You can prevent lumigo from sending keys that answer specific regexes by defining `LUMIGO_BLACKLIST_REGEX=["regex1", "regex2"]`. By default, we use the default regexes `[".*pass.*", ".*key.*"]`. All the regexes are case-insensitive.
+* Similarly, you can prevent lumigo from sending the entire headers and body of specific domains using `LUMIGO_DOMAINS_SCRUBBER=[".*secret.*"]`, or by specify the list of regexes with the key `domains_scrubber` in the tracer's decorator. By default, we will use `["secretsmanager\..*\.amazonaws\.com", "ssm\..*\.amazonaws\.com", "kms\..*\.amazonaws\.com"]`.
```
Comment (nirLumigo) [inline README.md]: Mention that `LUMIGO_DOMAINS_SCRUBBER` is an environment variable.

File: src/lumigo_tracer/utils.py (hunk 2 above)
Comment (saartochner) [reply src/lumigo_tracer/utils.py]: I don't think so 😮 
I think that the more flexible parameter should override less flexible, and environment variable is much more flexible than code 😮

File: src/lumigo_tracer/utils.py (hunk 4 above)
Comment (saartochner) [reply src/lumigo_tracer/utils.py]: yep,
and that's what I demand in the readme...

do you have another idea?

File: src/lumigo_tracer/utils.py (hunk 2 above)
Comment (saartochner) [reply src/lumigo_tracer/utils.py]: you won

**Pull request merged**